lib/reverse_words.rb

```diff
@@ -1,6 +1,36 @@
 # A method to reverse each word in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def reverse_words(my_words)
-  raise NotImplementedError
-end
+  return nil if my_words.nil?
+
+  start_word = 0
+  end_word = 0
+  i = 0
+  
+  unless my_words == nil
```

Comment:
You shouldn't need this check here if you have already checked for nil on line 5.